internal/publisher/redis_stream_publisher.go

```diff
@@ -0,0 +1,56 @@
+package publisher
+
+import (
+	"context"
+	"encoding/json"
+
+	"github.com/KyberNetwork/evmlistener/pkg/types"
+	"go.uber.org/zap"
+)
+
+type RedisStreamPublisher struct {
+	client Client
+	config Config
+	logger *zap.SugaredLogger
+}
+
+func NewRedisStreamPublisher(client Client, cfg Config) *RedisStreamPublisher {
+	return &RedisStreamPublisher{
+		client: client,
+		config: cfg,
+	}
+}
+
+func (p *RedisStreamPublisher) Publish(ctx context.Context, msg types.Message) error {
+	p.logger.Infow("Publish message to queue",
+		"topic", p.config.Topic,
+		"numRevertedBlocks", len(msg.RevertedBlocks),
+		"numNewBlocks", len(msg.NewBlocks))
+
+	data, err := p.packMsgData(msg)
+	if err != nil {
+		p.logger.Errorf("error on packing message to publish: %v", err)
+
+		return err
+	}
+
+	err = p.client.Publish(ctx, p.config, data, nil)
+	if err != nil {
+		p.logger.Errorf("error publish to redis stream: %v", err)
+
+		return err
+	}
+
+	return nil
+}
+
+func (p *RedisStreamPublisher) packMsgData(msg types.Message) ([]byte, error) {
```

Comment:
This is a new version we can use protobuf for encoding and compress the data to save space.

Reply:
Yes, it is a breaking change, the current deployment will not be affected.

Reply:
So I suggest just let it be, and we can change it in another PR, since it's not in the scope of this PR

Reply:
Ok